geoapi/tasks/external_data.py

```diff
@@ -375,90 +375,141 @@ def import_files_recursively_from_path(session, tenant_id: str, userId: int, sys
+
 @app.task()
-def refresh_observable_projects():
+def refresh_watch_content_projects():
     """
-    Refresh all observable projects
+    Refresh users for all projects where watch_content is True
     """
+    start_time = time.time()
+    with create_task_session() as session:
+        try:
+            logger.info("Starting to refresh all projects where watch_content is True")
+            projects_with_watch_content = session.query(Project).filter(Project.watch_content.is_(true())).all()
+            for i, project in enumerate(projects_with_watch_content):
+                try:
+                    importing_user = _get_user_with_valid_token(project)
+
+                    if importing_user is None:
+                        logger.error(f"Unable to watch content of project"
+                                     f" ({i}/{len(projects_with_watch_content)}): observer:{importing_user} "
+                                     f"system:{project.system_id} path:{project.system_path} project:{project.id} "
+                                     f"watch_content:{project.watch_content}: No user with an active token found. "
+                                     f"So we are skipping (i.e. no update of users or importing of watched "
+                                     f"content)")
+                        continue
+
+                    # perform the importing
+                    if project.watch_content:
+                        logger.info(f"Refreshing content of project ({i}/{len(projects_with_watch_content)}): "
+                                    f"observer:{importing_user} system:{project.system_id} path:{project.system_path} "
+                                    f"project:{project.id} watch_content:{project.watch_content}")
+                        import_files_recursively_from_path(session, project.tenant_id,
+                                                           importing_user.id, project.system_id,
+                                                           project.system_path, project.id)
+                except Exception:  # noqa: E722
+                    logger.exception(f"Unhandled exception when importing for project:{project.id}. "
+                                     "Performing rollback of current database transaction")
+                    session.rollback()
+            total_time = time.time() - start_time
+            logger.info("refresh_watch_content_projects completed. "
+                        "Elapsed time {}".format(datetime.timedelta(seconds=total_time)))
+        except Exception:  # noqa: E722
+            logger.error("Error when trying to get list of projects where watch_content is True; "
+                         "this is unexpected and should be reported "
+                         "(i.e. https://jira.tacc.utexas.edu/browse/WG-131).")
+            raise
+
 
-    # TODO refactor to consider scaling issues; see https://tacc-main.atlassian.net/browse/WG-47
+@app.task()
+def refresh_watch_users_projects():
```

Comment:
tic comment #4: What about `refresh_projects_watch_users()`?

Reply:
fixed in e6abc452696186e35253c9b111794b02ef0506ad